src/System.Text.Json/src/Resources/Strings.resx

```diff
@@ -177,6 +177,12 @@
   <data name="InvalidHexCharacterWithinString" xml:space="preserve">
     <value>'{0}' is not a hex digit following '\u' within a JSON string. The string should be correctly escaped.</value>
   </data>
+  <data name="JsonDocumentDoesNotSupportComments" xml:space="preserve">
+    <value>Comments cannot be incorporated into JsonDocument, only the Skip and Allow comment handling modes are supported.</value>
```

Comment:
"Comments cannot be incorporated into JsonDocument" => "Comments cannot be stored in a JsonDocument"?

Also... why the limitation?

Reply:
Mainly, complications in storage and retrieval; and then policy decisions... like what is the array count for `[ /* 0, */ 1, 2, 3, 4 ]`?  What would elem[0] return, and would the iterator produce it?  The object enumerator can't make a property for a free-floating comment, so it would have to skip it, or throw...

Once nothing is capable of reporting they're there, we shouldn't store them.  And once we're not storing them we should be honest that we simply didn't store them by saying Allow isn't respected, which means Skip and Disallow (and that "Allow" in this message is wrong :)).